docs/src/tutorials/entry-tutorial.md

```diff
@@ -0,0 +1,1085 @@
+# Entry-level Tutorial on the Model Checker
+
+**Difficulty: Blue trail -- Easy**
+
+In this tutorial, we show how to turn an implementation of binary search into a
+TLA+ specification. This implementation is known to have an out-of-bounds
+error, which once existed in Java, see [Nearly All Binary Searches and
+Mergesorts are Broken][] by Joshua Bloch (2006). Our goal is not to write a
+specification of an abstract binary search algorithm. You can find such a
+specification and a proof in [Proving Safety Properties][] and [Binary search
+with a TLAPS proof][] by [Leslie Lamport][] (2019).
+
+This tutorial is written under the assumption that the reader does not have any
+knowledge of TLA+ and Apalache. Since we are not diving into protocol and
+algorithm specifications too quickly, this is a nice example to start with.
+
+Sometimes, we refer to the model checker TLC in this text. TLC is the standard
+model checker for TLC, which was introduced in the late 90es.  If you are new
+to TLA+ and want to learn more about TLC, check the [TLC][] project and the
+[TLA+ Video Course][] by Leslie Lamport.  If you are an experienced TLC user,
+you will find this tutorial helpful too, as it demonstrates the strong points
+of Apalache.
+
+We discover that the same bug not only causes the out-of-bounds error, but also
+prevents the algorithm from terminating in `log2(high - low + 1)`, as it is
+usually expected.
+
+## Related documents
+
+ - [Tutorial on Snowcat][] shows how to write type annotations.
+ - [TLA+ Cheatsheet in HTML][] summarizes the common TLA+ constructs.  If you
+   prefer the printable version in pdf, check the [Summary of TLA+][].
+
+## Setup
+
+We assume that you have Apalache installed. If not, check the manual page on
+[Apalache installation][]. The minimal required version is 0.22.0.
+
+## Running example: Binary search
+
+We are not going to explain the idea of binary search in this tutorial. If you
+need more context on this, check the Wikipedia page on the [Binary search
+algorithm][]. Let's jump straight in the Java code that is given in [Nearly All
+Binary Searches and Mergesorts are Broken][]:
+
+```java
+1:     public static int binarySearch(int[] a, int key) {
+2:         int low = 0;
+3:         int high = a.length - 1;
+4:
+5:         while (low <= high) {
+6:             int mid = (low + high) / 2;
+7:             int midVal = a[mid];
+8:
+9:             if (midVal < key)
+10:                 low = mid + 1
+11:             else if (midVal > key)
+12:                 high = mid - 1;
+13:             else
+14:                 return mid; // key found
+15:         }
+16:         return -(low + 1);  // key not found.
+17:     }
+```
+
+Fast forward, as was found by Joshua Bloch, the expression in line 6 may throw
+an out of bounds exception, due to an integer overflow. This is because `low`
```

Comment:
```suggestion
Fast forward, as was found by Joshua Bloch, the addition in line 6 may throw
an out of bounds exception at line 7, due to an integer overflow. This is because `low`
```